bin/hash2kmer.py

```diff
@@ -0,0 +1,187 @@
+    p.add_argument('--output-kmers', type=str, default=None,
+                   help='save matching kmers to this file.')
+    p.add_argument('-k', '--ksize', type=int, required=True)
+    p.add_argument(
+        '--input-is-protein', action='store_true',
+        help='Consume protein sequences - no translation needed.'
+    )
+    p.add_argument(
+        '--first', action='store_true',
+        help='Return only the first instance of the found k-mer(s) and ' \
+             'sequence from each provided sequence file. Useful if you are ' \
+             'searching for only one k-mer'
+    )
+    add_construct_moltype_args(p)
+    args = p.parse_args()
+
+    # set up the outputs.
+    seqout_fp = None
+    if args.output_sequences:
+        seqout_fp = open(args.output_sequences, 'wt')
+
+    kmerout_fp = None
+
+    if not (seqout_fp or kmerout_fp):
+        error("No output options given!")
+        return(-1)
+
+    if args.output_kmers:
+        kmerout_fp = open(args.output_kmers, 'wt')
+        kmerout_w = csv.writer(kmerout_fp)
+        kmerout_w.writerow(['kmer', 'hashval'])
+
+    # Ensure that protein ksizes are divisible by 3
+    if (args.protein or args.dayhoff or args.hp) and not args.input_is_protein:
+        if args.ksize % 3 != 0:
+            error('protein ksizes must be divisible by 3, sorry!')
+            error('bad ksizes: {}', ", ".join(args.ksize))
+            sys.exit(-1)
+
+    # load in all the hashes
+    hashes = set()
+    for line in open(args.hashfile, 'rt'):
+        hashval = int(line.strip())
+        hashes.add(hashval)
+
+    moltype = calculate_moltype(args)
+
+    if not hashes:
+        error("ERROR, no hashes loaded from {}!", args.hashfile)
+        return -1
+
+    notify('loaded {} distinct hashes from {}', len(hashes), args.hashfile)
+
+    # now, iterate over the input sequences and output those that overlap
+    # with hashes!
+    n_seq = 0
+    n = 0 # bp loaded
+    m = 0 # bp in found sequences
+    p = 0 # number of k-mers found
+    found_kmers = {}
+    watermark = NOTIFY_EVERY_BP
+    for filename in args.seqfiles:
+        m, n = get_matching_hashes_in_file(
+            filename, args.ksize, moltype, args.input_is_protein, hashes,
+            found_kmers, m, n, n_seq, seqout_fp, watermark, args.first)
+        if args.first and m > 0:
+            break
+
+    if seqout_fp:
+        notify('read {} bp, wrote {} bp in matching sequences', n, m)
+
+    if kmerout_fp and found_kmers:
+        for kmer, hashval in found_kmers.items():
+            kmerout_w.writerow([kmer, str(hashval)])
+        notify('read {} bp, found {} kmers matching hashvals', n,
+               len(found_kmers))
+
+
+def get_matching_hashes_in_file(filename, ksize, moltype, input_is_protein,
+                                hashes, found_kmers, m, n,
+                                n_seq, seqout_fp, watermark, first=False):
+    for record in screed.open(filename):
```

Comment:
I guess this works for now. But it would be great if they are in a repo and unit tested.